LibreNMS/OS/Traits/UcdResources.php

```diff
@@ -49,4 +50,68 @@ public function discoverProcessors()
             ),
         ];
     }
+
+    public function discoverMempools()
+    {
+        $mempools = collect();
+        $data = snmp_get_multi($this->getDeviceArray(), [
+            'memTotalSwap.0',
+            'memAvailSwap.0',
+            'memTotalReal.0',
+            'memAvailReal.0',
+            'memBuffer.0',
+            'memCached.0',
+        ], '-OQUs', 'UCD-SNMP-MIB');
+
+        if ($this->oidValid($data, 'memTotalReal') && ($this->oidValid($data, 'memAvailReal') || $this->oidValid($data, 'memSysAvail'))) {
+            $mempools->push((new Mempool([
+                'mempool_index' => 1,
+                'mempool_type' => 'ucd',
+                'mempool_class' => 'system',
+                'mempool_precision' => 1024,
+                'mempool_descr' => 'Physical memory',
+                'mempool_free_oid' => '.1.3.6.1.4.1.2021.4.6.0',
+            ]))->fillUsage(null, $data[0]['memTotalReal'], $data[0]['memAvailReal'] + $data[0]['memBuffer'] + $data[0]['memCached']));
```

Comment:
shall we consider using 'memSysAvail' when it's present?

Reply:
@ibigbug Check now, you commented before I pushed changes to include memSysAvail.